### PR TITLE
[KS-193] Pass MercuryTriggerService to Mercury Transmitter

### PR DESCRIPTION
**Pull request by bolekk**
1. Add EnableTriggerCapability flag to Relay config
2. Create MercuryTriggerService lazily, on the first call to NewMercuryProvider()
3. Make it available in the Transmitter (no-op for now)

### CODE REVIEW

File: core/services/ocr2/delegate.go
Comment (bolekk) [inline core/services/ocr2/delegate.go]: @cedric-cordenier suggestions?

Comment (cedric-cordenier) [reply core/services/ocr2/delegate.go]: @bolekk We should add the bridge via the relayer constructor; I'm working on the signatures PR which will add this

File: core/services/relay/evm/evm.go
Comment (cedric-cordenier) [inline core/services/relay/evm/evm.go]: @bolekk Per my DMs, pass there registry via the RegistryOpts here.

File: core/services/relay/evm/evm.go
Comment (cedric-cordenier) [inline core/services/relay/evm/evm.go]: This is nice and simple so +1

File: core/services/chainlink/application.go
Comment (cedric-cordenier) [inline core/services/chainlink/application.go]: Please use `coretypes.CapabilitiesRegistry` here rather than the concrete type (`chainlink-common/pkg/types/core`) as that will make loopification easy

File: core/services/chainlink/relayer_factory.go
Comment (cedric-cordenier) [inline core/services/chainlink/relayer_factory.go]: Same here

**Review by cedric-cordenier — DISMISSED** [review]
Some minor nits but otherwise LGTM

**Review by cedric-cordenier — DISMISSED** [review]

File: core/capabilities/syncer.go
Comment (samsondav) [inline core/capabilities/syncer.go:131]: Seems weird to have mocks explicitly instantiated in application code, can we not use dependency injection here?

**Review by samsondav — APPROVED** [review]

File: core/capabilities/syncer.go
Comment (bolekk) [reply core/capabilities/syncer.go:131]: This Syncer object is soon going to be replaced with a real one.

File: core/services/relay/evm/evm.go
Comment (krehermann) [inline core/services/relay/evm/evm.go:263]: does this need to be guarded with some sort of singleton logic or is Add idempotent? NewMercuryProvider can be  called in multiple times

Comment (bolekk) [reply core/services/relay/evm/evm.go:263]: Yeah this is guarded by "r.triggerCapability == nil" check and I believe NewMercuryProvider() will not be called in parallel (because delegate.ServicesForSpec() isn't)

File: core/services/relay/evm/mercury/transmitter.go
Comment (krehermann) [inline core/services/relay/evm/mercury/transmitter.go:280]: so many args is yellow flag. i'd prefer to see them consolidated into an opt or config struct and constructor handle/validate as needed. passing `nil` in constructors is a signal that we are really passing options

Comment (bolekk) [reply core/services/relay/evm/mercury/transmitter.go:280]: I'd rather leave that to code owners here, I'm only adding the trigger service and don't fully understand the consequences of refactoring.

**Review by krehermann — APPROVED** [review]

**Pull request merged**
